website/content/api-docs/system/config-ui-custom-messages.mdx

```diff
@@ -0,0 +1,242 @@
+---
+layout: api
+page_title: /sys/config/ui/custom-messages - HTTP API
+description: The '/sys/config/ui/custom-messages' endpoint configures the custom messages that the UI displays prior to and immediately following login operations.
+---
+
+# `/sys/config/ui/custom-messages`
+
+The `/sys/config/ui/custom-messages` endpoint is used to list, create, read, update, or delete custom messages.
+
+## List Custom Messages
+
+This endpoint returns a list of all custom messages that exist.
+
+| Method | Path                              |
+| :----- | :-------------------------------- |
+| `LIST` | `/sys/config/ui/custom-messages/` |
+
+### Parameters
+
+- `active` `(bool: <optional>)` - A flag indicating whether to only include custom messages whose active property is set to the value specified in the results. If this parameter is omitted, the results include both active and inactive custom messages.
+
+- `authenticated` `(bool: <optional>)` - A flag indicating whether to only include custom messages whose authenticated property is set to the value specified in the results. If this parameter is omitted, the results include both authenticated and unauthenticated custom messages.
+
+- `type` `(string: <optional>)` - A flag that can only be set to either `banner` or `modal`. It indicates whether to only include custom messages whose type property is set to the value specified in the results. If this parameter is omitted, the results include both types of custom messages.
+
+### Sample request
+
+```shell-session
+$ curl --header "X-Vault-Token: ..." \
+    --request LIST \
+    http://127.0.0.1:8200/v1/sys/config/ui/custom-messages?active=true&authenticated=false&type=modal
+```
+
+### Sample response
+
+```json
+{
+  "data": {
+    "key_info": {
+      "01234567-89ab-cdef-0123-456789abcdef": {
+        "title": "Post-login Advisory",
+        "type": "modal",
+        "authenticated": false,
+        "start_time": "2024-01-01T00:00:00.000000000Z",
+        "end_time": null
+      }
+    },
+    "keys": [
+      "01234567-89ab-cdef-0123-456789abcdef"
+    ]
+  }
+}
+```
+
+## Create Custom Message
+
+This endpoint allows the creation of a new custom message. Vault allows up to 100 custom messages
+per namespace. If this endpoint is called to create a custom message when the maximum number of
+custom messages already exists, an error will be returned.
```

Comment:
I'm not sure if this should be mentioned in the docs - if a message is created on the parent namespace, it will show up for child namespaces as well.

Reply:
That's a good point. I meant to cover that but forgot. I think the better place to explain that aspect of the feature is in both the _internal-ui-authenticated-messages.mdx_ and _internal-ui-unauthenticated-messages.mdx_ files.